### DISCUSSION TITLE
Checkboxes

### DISCUSSION THREAD

**Discussion by jrbarnes9** [ONSdigital/design-system#13] [Feedback]
Use this place to discuss the [checkboxes component](https://ons-design-system.netlify.com/components/checkboxes) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Currently there is no way to pass attributes to the checkboxes. It would be good to pass attributes to checkboxes like on inputs.

**Comment by hamishtaplin** [comment]
The examples show "select all that apply" as the `legend` whereas it should be a `.field__label`.

**Comment by jrbarnes9** [comment]
Screen reader doesn't read out revealed text input field label or content when selecting 'other' checkbox option.

**Comment by jrbarnes9** [comment]
> The examples show "select all that apply" as the `legend` whereas it should be a `.field__label`.

Covered in: #271

**Comment by jrbarnes9** [comment]
> Screen reader doesn't read out revealed text input field label or content when selecting 'other' checkbox option.

#272

**Comment by bameyrick** [comment]
The value parameter for the "other" text input isn't passed to the input macro so the field cannot be repopulated.

**Comment by glancyons** [comment]
The code for basic checkboxes:
https://ons-design-system.netlify.com/components/checkboxes/
 does not validate using
https://validator.w3.org/nu/

Error is:
Element p not allowed as child of element span in this context.

**Comment by jrbarnes9** [comment]
Min width of text input (19.5rem) is different to min width of checkboxes and radios (20rem)

<img width="449" alt="Screenshot 2020-09-03 at 08 26 38" src="https://user-images.githubusercontent.com/43346934/92084489-622eeb80-edbf-11ea-9d12-0ef0a69b7c66.png">

**Reply by jrbarnes9** [reply]
#951 resolved

**Comment by jrbarnes9** [comment]
Remove `margin-bottom` from `fieldset__legend` if `legendIsPageTitle` is used.  

The `h1` already has the required margin

**Reply by boxadesign** [reply]
https://github.com/ONSdigital/design-system/pull/1783

**Comment by jrbarnes9** [comment]
A selected checkbox/radio input border should be red when wrapped in error

Currently:
![Screenshot 2021-09-14 at 17 06 49](https://user-images.githubusercontent.com/43346934/133293768-0a28f1f4-0d0d-4329-936f-d0d4b8216fa0.png)

Should be:
![Screenshot 2021-09-14 at 17 07 43](https://user-images.githubusercontent.com/43346934/133293901-64968b3f-d464-4a64-add2-de5c21336199.png)

**Reply by daz2048** [reply]
Checkboxes don't work with latest template, without CSS style.
They also need a 'name' which is missing from example. 
So working minimum would be: 

    <div class="ons-grid__col ons-col-3@m ">
      <div class="ons-pl-grid-col">
	{{
        onsCheckboxes({
            "checkboxesLabel": "Clearing Status:",
            "dontWrap": true,
            "checkboxes": [
                {
                    "id": "clear",
                    "name": "statuses-clear",
                    "label": {
                        "text": "CLEAR"
                    },
                    "value": "clear"
                }
}}
</div>
</div>   

Can we please update example on https://ons-design-system.netlify.app/components/checkboxes/

**Reply by jrbarnes9** [reply]
Raised #1787

**Reply by boxadesign** [reply]
@dariusz22p Can you provide more info on what you mean "don't work". Is this a specific variant?

**Comment by jrbarnes9** [comment]
Borderless checkboxes should always have a white background
![Screenshot 2021-12-14 at 15 21 31](https://user-images.githubusercontent.com/43346934/146027826-7d951610-3f7e-43f6-9c9d-561cf7e36e19.png)

**Reply by jrbarnes9** [reply]
Raised #1800

**Comment by jrbarnes9** [comment]
'Select all' button submits form

**Reply by jrbarnes9** [reply]
Raised #2110

**Comment by armstrongb** [comment]
### Description
The use of checkboxes in filters requires the addition of a disabled state for items which return zero result counts.

- `$Grey35` can safely be used for the checkbox and label as [WCAG2.1 states](https://www.w3.org/TR/WCAG21/#contrast-minimum) "disabled interface need not meet contrast minimum" (see incidental)
- `aria-disabled` could be applied to the `label` and `input` to supplement the standard `<input ... disabled />` [MDN reference](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled)
- cursor should be set to `cursor: not-allowed;` for label and input.

As seen in 

![Publications filter selected (disabled zero results)](https://user-images.githubusercontent.com/22910336/163198948-3455e1cc-6485-4237-8a90-375da1c35a07.png)

and
<img width="197" alt="Screenshot 2022-04-13 at 11 09 40" src="https://user-images.githubusercontent.com/22910336/163197697-571e75be-04c1-4d8c-ba4d-f84c34087c78.png">

@jondewijones following this item for use in search.

**Reply by jrbarnes9** [reply]
Raised #2141

**Comment by jrbarnes9** [comment]
Tabbing onwards from visible text input put focus on first input rather than the next focusable element.

![Screenshot 2022-08-05 at 12 14 08](https://user-images.githubusercontent.com/43346934/183066379-69207d1f-f6d2-41bd-aaee-3b845d460533.png)

**Reply by jrbarnes9** [reply]
Raised #2375

**Comment by AlanHONS** [comment]
Requesting enhancement to Checkbox component to allow 'Revealed Textarea' with character count on Checkboxes. 
In the same way this was enabled for Radio buttons as part of 70.0.4 - See https://github.com/ONSdigital/design-system/discussions/56#discussioncomment-9044812

Confirm that we have a definite need for this to deliver current design for RAS Replacement IDS Project Approval UI flow

![{7B04C81B-6DDF-4185-9699-010EF117143B}](https://github.com/user-attachments/assets/1160824f-19df-44ed-8120-7801a07449fd)
